platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/BirthRoundStateMigration.java

```diff
@@ -0,0 +1,115 @@
+     * time to migrate, or if the migration has already been completed.
+     *
+     * @param initialState the initial state the platform is starting with
+     * @param ancientMode  the current ancient mode
+     * @param appVersion   the current application version
+     */
+    public static void modifyStateForBirthRoundMigration(
+            @NonNull final SignedState initialState,
+            @NonNull final AncientMode ancientMode,
+            @NonNull final SoftwareVersion appVersion) {
+
+        if (ancientMode == AncientMode.GENERATION_THRESHOLD) {
+            if (initialState.getState().getPlatformState().getFirstVersionInBirthRoundMode() != null) {
+                throw new IllegalStateException(
+                        "Cannot revert to generation mode after birth round migration has been completed.");
+            }
+
+            logger.info(
+                    STARTUP.getMarker(), "Birth round state migration is not yet needed, still in generation mode.");
+            return;
+        }
+
+        final State state = initialState.getState();
+        final PlatformState platformState = state.getPlatformState();
+
+        final boolean alreadyMigrated = platformState.getFirstVersionInBirthRoundMode() != null;
+        if (alreadyMigrated) {
+            // Birth round migration was completed at a prior time, no action needed.
+            logger.info(STARTUP.getMarker(), "Birth round state migration has already been completed.");
+            return;
+        }
+
+        final long lastRoundBeforeMigration =
+                initialState.getState().getPlatformState().getRound();
+
+        final ConsensusSnapshot consensusSnapshot = Objects.requireNonNull(platformState.getSnapshot());
+        final List<MinimumJudgeInfo> judgeInfoList = consensusSnapshot.getMinimumJudgeInfoList();
+        final long lowestJudgeGenerationBeforeMigration =
+                judgeInfoList.getLast().minimumJudgeAncientThreshold();
+
+        logger.info(
+                STARTUP.getMarker(),
+                "Birth round state migration in progress. First version in birth round mode: {}, "
+                        + "last round before migration: {}, lowest judge generation before migration: {}",
+                appVersion,
+                lastRoundBeforeMigration,
+                lowestJudgeGenerationBeforeMigration);
+
+        platformState.setFirstVersionInBirthRoundMode(appVersion);
+        platformState.setLastRoundBeforeBirthRoundMode(lastRoundBeforeMigration);
+        platformState.setLowestJudgeGenerationBeforeBirthRoundMode(lowestJudgeGenerationBeforeMigration);
+
+        final List<MinimumJudgeInfo> modifiedJudgeInfoList = new ArrayList<>(judgeInfoList.size());
+        for (final MinimumJudgeInfo judgeInfo : judgeInfoList) {
+            modifiedJudgeInfoList.add(new MinimumJudgeInfo(
+                    judgeInfo.round(),
+                    initialState.getState().getPlatformState().getRound()));
```

Comment:
You can use `lastRoundBeforeMigration` local variable instead of calling `initialState.getState().getPlatformState().getRound()` again

Reply:
fixed